test/unit_tests/26_host_nested_loop/aieDoubleOutBuffer.mlir

```diff
@@ -0,0 +1,90 @@
+//===- aie.mlir ------------------------------------------------*- MLIR -*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// Copyright (C) 2022, Advanced Micro Devices, Inc.
+//
+//===----------------------------------------------------------------------===//
+
+// REQUIRES: valid_xchess_license
+// RUN: aiecc.py -j4 --sysroot=%VITIS_SYSROOT% --host-target=aarch64-linux-gnu %s -I%aie_runtime_lib%/ %extraAieCcFlags% %aie_runtime_lib%/test_library.cpp %S/test.cpp -o test.elf
+// RUN: %run_on_board ./test.elf
+
+module @host_loop {
+    %tile34 = AIE.tile(3, 3)
+    %tile70 = AIE.tile(7, 0)
+
+    func.func @evaluate_condition(%argIn : i32) -> (i1) {
+        %true = arith.constant 1 : i1
+        return %true : i1
+    }
+
+    func.func @payload(%argIn : i32) -> (i32) {
+        %next = arith.constant 1 : i32
+        return %next : i32
+    }
+
+    %ext_buf70_in  = AIE.external_buffer {sym_name = "ddr_test_buffer_in"}: memref<256xi32> 
+    %ext_buf70_out1 = AIE.external_buffer {sym_name = "ddr_test_buffer_out1"}: memref<64xi32>
+    %ext_buf70_out2 = AIE.external_buffer {sym_name = "ddr_test_buffer_out2"}: memref<64xi32> 
+
+    %objFifo_in = AIE.objectFifo.createObjectFifo(%tile70, {%tile34}, 1) : !AIE.objectFifo<memref<256xi32>>
+    %objFifo_out = AIE.objectFifo.createObjectFifo(%tile34, {%tile70}, 1) : !AIE.objectFifo<memref<64xi32>>
+    
+    AIE.objectFifo.registerExternalBuffers(%tile70, %objFifo_in : !AIE.objectFifo<memref<256xi32>>, {%ext_buf70_in}) : (memref<256xi32>)
+    AIE.objectFifo.registerExternalBuffers(%tile70, %objFifo_out : !AIE.objectFifo<memref<64xi32>>, {%ext_buf70_out1,%ext_buf70_out2}) : (memref<64xi32>,memref<64xi32>)
+
+ 
+    %core34 = AIE.core(%tile34) {
+        %c0 = arith.constant 0 : index
+        %c1 = arith.constant 1 : index
+        %c64 = arith.constant 64 : index
+        %c65 = arith.constant 65 : index
+        %c128 = arith.constant 128 : index
+        %init1 = arith.constant 1 : i32
+
+        %res = scf.while (%arg1 = %init1) : (i32) -> i32 {
```

Comment:
There's some specifics in the objectFifo lowering that don't work well with while loops, could you please replace this with a for loop instead?

Reply:
Thanks, fixed.